compareBuckets/Flusher.js

```diff
@@ -0,0 +1,136 @@
+const assert = require('assert');
+const events = require('events');
+const werelogs = require('werelogs');
+
+werelogs.configure({
+    level: 'info',
+    dump: 'error',
+});
+
+class Flusher {
+
+    // params:
+    // - stopAt: for tests
+    // - flusherTimeout: Flush timeout
+    // - interactive: if true flush is done manually
+    constructor(bucketName, persist, persistData, params) {
+        this.bucketName = bucketName;
+        this.persist = persist;
+        this.persistData = persistData;
+        // params
+        this.stopAt = -1;
+        this.flusherTimeout = 5000;
+        this.interactive = false;
+        if (params) {
+            if (params.stopAt !== undefined) {
+                this.stopAt = params.stopAt;
+            }
+            if (params.flusherTimeout !== undefined) {
+                this.flusherTimeout = params.flusherTimeout;
+            }
+            if (params.interactive !== undefined) {
+                this.interactive = params.interactive;
+            }
+        }
+        this.events = new events.EventEmitter();
+        this.addedQueue = [];
+        this.deletedQueue = [];
+        this.prevOffset = null;
+        this.offset = null;
+        this.pending = false;
+        this.logger = new werelogs.Logger('Daemon');
+    }
+
+    addEvent(item, offset, opType) {
+        // console.log('addEvent', this.bucketName, item, opType);
+        if (opType === 'put') {
+            this.addedQueue.push(item);
+        } else if (opType === 'del') {
+            this.deletedQueue.push(item);
+        } else {
+            assert(false);
```

Comment:
I would add a message to this assert at least, debugging would be very hard otherwise.